lib/config_yaml.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+# vim: set encoding=utf-8 tabstop=4 softtabstop=4 shiftwidth=4 expandtab
+#########################################################################
+# Copyright 2016-       Martin Sinn                         m.sinn@gmx.de
+#########################################################################
+#  This file is part of SmartHomeNG
+#  https://github.com/smarthomeNG/smarthome
+#  http://knx-user-forum.de/
+#
+#  SmartHomeNG is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  SmartHomeNG is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with SmartHomeNG. If not, see <http://www.gnu.org/licenses/>.
+#########################################################################
+
+import logging
+import collections
+import lib.shyaml as shyaml
+
+logger = logging.getLogger(__name__)
+
+
+def remove_comments(ydata, level=0):
+    '''
+    Removes comments from a dict or OrderedDict structure
+
+    :param ydata: configuration (sub)tree to work on
+    :param level: optional subtree level (used for recursion)
+    '''
+    level_keys = list(ydata.keys())
+    for key in level_keys:
+        if type(ydata[key]).__name__ in ['dict','OrderedDict']:
+            remove_comments(ydata[key], level+1)
+        else:
+            if key.startswith('comment'):
+                ydata.pop(key)
+
```

Comment:
Thats true. But for the moment I know no plugin that needs `comment`. I remove the comments, because the converter generates lots of comments. The converter picks up all `#`comments in the conf file and preserves them in the yaml file as `comment`attributes.